server/matcher/match.go

```diff
@@ -184,7 +184,7 @@ func (m *Matcher) Match(book Booker, queue []*OrderRevealed) (seed []byte, match
 			removed, ok := book.Remove(o.TargetOrderID)
 			if !ok {
 				// The targeted order might be down queue or non-existent.
-				log.Debugf("Failed to remove order %v set by a cancel order %v",
+				log.Debugf("Order %v not removed by a cancel order %v (target either non-existent or down queue in this epoch)",
```

Comment:
That's the idea.  Down queue means after shuffle of an epoch queue that contains both a trade and a cancel order targeting that trade, the target order would get processed after the cancel order, so the cancel order would not be able to cancel it since it hasn't been booked yet.